python/ops.py

```diff
@@ -63,6 +88,284 @@ def c2r_ifft2(x_fft):
           tf.signal.ifft2d(tf.transpose(x_fft, [0, 3, 1, 2])), [0, 2, 3, 1]))
 
 
+
+
+def local_absolute_deviation(rgb):
+  """Compute a Local Absolute Deviation in sliding 3x3 window fashion.
+
+  This functions computes the edge signal of a given rgb image.
+
+  Args:
+    rgb: RGB image (float32) with the shape of [batch_size, height, width,
+      channels].
+
+  Returns:
+    rgb_edge: RGB image (float32) with the shape of [batch_size, height, width,
+    channels] of the corresponding edge signals of each color channel in the rgb
+    input image.
+  """
+
+  # Padding the image and the mask (if given) before computing the image's
+  # edges
+  for c in range(3):  # for each color channel
```

Comment:
for loops are usually signs that there is a better way to organize your computation in tensorflow. Can you write this such that you're computing edge signals for all channels at once?

Reply:
Padding with 'SYMMETRIC' is only provided for single-channel images.

Reply:
I don't see any mention of single channel images in tf.pad. You want to add padding of 1 pixel in top, left, right and bottom , right? Can you check this colab, scroll to the end, does this do the trick?
```
paddings = tf.constant([[0,0], [1,1], [1,1], [0,0]], dtype='int32') # pad +1 in width and height, but not in channels nor batch_size
rgb_padded = tf.pad(rgb, paddings, mode='SYMMETRIC')
```
https://colab.sandbox.google.com/drive/18ShK8zaA5XpLJKcmxz59KUOFwePS47VE#scrollTo=1tAK80maJrr-

I think we should have zero for loops within tensorflow operators, unless they're absolutely necessary.

Reply:
This works. Many thanks!